api/integrations/github/github.py

```diff
@@ -81,22 +89,37 @@ def generate_body_comment(
     last_updated_string = LAST_UPDATED_FEATURE_TEXT % (
         datetime.datetime.now().strftime("%dth %b %Y %I:%M%p")
     )
-    updated_text = UPDATED_FEATURE_TEXT % (name)
 
-    result = updated_text if is_update else LINK_FEATURE_TEXT % (name)
+    result = UPDATED_FEATURE_TEXT % (name) if is_update else LINK_FEATURE_TITLE % (name)
+    last_segment_name = ""
+    if len(feature_states) > 0 and not feature_states[0].get("segment_name"):
+        result += FEATURE_TABLE_HEADER
 
-    # if feature_states is None:
     for v in feature_states:
         feature_value = v.get("feature_state_value")
         feature_value_type = v.get("feature_state_value_type")
+        tab = "segment-overrides" if v.get("segment_name") is not None else "value"
+        environment_link_url = FEATURE_ENVIRONMENT_URL % (
+            get_current_site_url(),
+            project_id,
+            v.get("environment_api_key"),
+            feature_id,
+            tab,
+        )
+        if v.get("segment_name") is not None and v["segment_name"] != last_segment_name:
+            result += "\n" + LINK_SEGMENT_TITLE % (v["segment_name"])
+            last_segment_name = v["segment_name"]
+            result += FEATURE_TABLE_HEADER
+        table_row = FEATURE_TABLE_ROW % (
+            v["environment_name"],
+            environment_link_url,
+            "✅ Enabled" if v["feature_value"] else "❌ Disabled",
+            feature_value if feature_value else "",
+            feature_value_type,
```

Comment:
I don't think we should surface feature value type here:

1. We plan to move away from feature value types to only support strings in the future.
2. String value type awkwardly gets shown as "Unicode" which, ideally, should be replaced by "String" but it's messy and I'd rather skip the whole ordeal.

Reply:
Done.